### PR TITLE
feat(ts): convert poweredBy widget

### PR DESCRIPTION
**Pull request by shortcuts**
**Summary**

Converts `poweredBy` widget to TypeScript.

### CODE REVIEW

File: src/widgets/powered-by/__tests__/powered-by-test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+"The \`container\` option is required.
+
+See documentation: https://www.algolia.com/doc/api-reference/widgets/powered-by/js/"
+`);
+  });
+});
+
+describe('poweredBy', () => {
+  let widget: ReturnType<typeof poweredBy>;
+  let container: string | HTMLElement;
```
Comment (francoischalifour) [inline src/widgets/powered-by/__tests__/powered-by-test.ts:33]: It's never a `string` in the test, is it?

```suggestion
  let container: HTMLElement;
```

File: src/widgets/powered-by/__tests__/powered-by-test.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  it('renders only once at init', () => {
+    widget.render!(createRenderOptions({ helper }));
+    widget.render!(createRenderOptions({ helper }));
+
+    const firstRender = render.mock.calls[0][0] as VNode<PoweredByProps>;
+    const firstContainer = render.mock.calls[0][1];
+
+    expect(render).toHaveBeenCalledTimes(1);
+    expect(firstRender.props).toMatchInlineSnapshot(`
```
Comment (francoischalifour) [inline src/widgets/powered-by/__tests__/powered-by-test.ts:62]: We could use `toEqual` to read it more easily.

File: src/widgets/powered-by/powered-by.tsx [hunk 3]
```diff
@@ -29,36 +37,48 @@ const renderer = ({ containerNode, cssClasses }) => (
...
- * @example
- * search.addWidgets([
- *   instantsearch.widgets.poweredBy({
- *     container: '#poweredBy-container',
- *     theme: 'dark',
- *   })
- * ]);
- */
-export default function poweredBy(widgetParams) {
+export type PoweredByCSSClasses = {
```
Comment (francoischalifour) [inline src/widgets/powered-by/powered-by.tsx:40]: We can get the required form in the component and do this here:

```ts
type PoweredByCSSClasses = Partial<PoweredByComponentCSSClasses>
```

Comment (Haroenv) [reply src/widgets/powered-by/powered-by.tsx:40]: or even required here, and partial https://github.com/algolia/instantsearch.js/pull/4756/files#diff-a1c4756cb305947c3149d511a59b64ba6bd6f268eb5eed4f036eaa95d76d4e19R72

**Review by Haroenv — COMMENTED** [review]
same comments as Francois, fine for the rest!

File: src/widgets/powered-by/powered-by.tsx (hunk 3 above)
Comment (shortcuts) [reply src/widgets/powered-by/powered-by.tsx:40]: Does this make sense to you? https://github.com/algolia/instantsearch.js/pull/4756/commits/0b93a236f1c4c693a3f085aadb0acf04fedf5090

File: src/widgets/powered-by/__tests__/powered-by-test.ts (hunk 2 above)
Comment (shortcuts) [reply src/widgets/powered-by/__tests__/powered-by-test.ts]: Good catch, done in https://github.com/algolia/instantsearch.js/pull/4756/commits/5d1b13468290b1f3ecbaea5028b7d529b43dfc6f

File: src/components/PoweredBy/PoweredBy.tsx [hunk 4]
```diff
@@ -1,31 +1,27 @@
...
-  link: string;
-  logo: string;
-};
-
-type Props = {
+export type PoweredByProps = {
   url: string;
   theme: string;
-  cssClasses: CSSClasses;
+  cssClasses: PoweredByCSSClasses;
```
Comment (Haroenv) [inline src/components/PoweredBy/PoweredBy.tsx:10]:
```suggestion
  cssClasses: PoweredByComponentCSSClasses;
```

File: src/components/PoweredBy/PoweredBy.tsx [hunk 5]
```diff
@@ -1,31 +1,27 @@
...
+import { PoweredByCSSClasses } from '../../widgets/powered-by/powered-by';
 
-type CSSClasses = {
-  root: string;
-  link: string;
-  logo: string;
-};
-
-type Props = {
+export type PoweredByProps = {
```
Comment (Haroenv) [inline src/components/PoweredBy/PoweredBy.tsx:7]:
```suggestion
export type PoweredByComponentCSSClasses = {
  [TClassName in keyof PoweredByCSSClasses]: string;
};

export type PoweredByProps = {
```

File: src/components/PoweredBy/PoweredBy.tsx [hunk 6]
```diff
@@ -1,31 +1,27 @@
...
   url: string;
   theme: string;
-  cssClasses: CSSClasses;
+  cssClasses: PoweredByCSSClasses;
 };
 
-const PoweredBy = ({ url, theme, cssClasses }: Props) => (
-  <div className={cssClasses.root}>
+const PoweredBy = ({ url, theme, cssClasses }: PoweredByProps) => (
+  <div className={cx(cssClasses.root)}>
```
Comment (Haroenv) [inline src/components/PoweredBy/PoweredBy.tsx:14]:
```suggestion
  <div className={cssClasses.root}>
```

File: src/components/PoweredBy/PoweredBy.tsx [hunk 7]
```diff
@@ -1,31 +1,27 @@
...
     <a
       href={url}
       target="_blank"
-      className={cssClasses.link}
+      className={cx(cssClasses.link)}
```
Comment (Haroenv) [inline src/components/PoweredBy/PoweredBy.tsx:18]:
```suggestion
      className={cssClasses.link}
```

File: src/components/PoweredBy/PoweredBy.tsx [hunk 8]
```diff
@@ -1,31 +1,27 @@
...
       aria-label="Search by Algolia"
       rel="noopener noreferrer"
     >
       <svg
         height="1.2em"
-        className={cssClasses.logo}
+        className={cx(cssClasses.logo)}
```
Comment (Haroenv) [inline src/components/PoweredBy/PoweredBy.tsx:24]:
```suggestion
        className={cssClasses.logo}
```

**Review by Haroenv — COMMENTED** [review]
no need to use cx twice :)

**Review by Haroenv — APPROVED** [review]

**Pull request merged**